src/mono/wasm/runtime/crypto-worker.ts

```diff
@@ -156,21 +195,54 @@ class LibraryChannel {
     public get_comm_buffer(): SharedArrayBuffer { return this.comm_buf; }
 
     public send_msg(msg: string): string {
-        if (Atomics.load(this.comm, this.STATE_IDX) !== this.STATE_IDLE) {
-            throw "OWNER: Invalid sync communication channel state. " + Atomics.load(this.comm, this.STATE_IDX);
+        try {
+            let state = Atomics.load(this.comm, this.STATE_IDX);
+            if (state !== this.STATE_IDLE)
+                console.log(`send_msg, waiting for idle now, ${state}`);
+            state = this.wait_for_state(pstate => pstate == this.STATE_IDLE, "waiting");
+
+            // FIXME: it won't get to this line for !IDLE
```

Comment:
@radical - thoughts on how to fix the 3 `FIXME`s in this file?

Reply:
`wait_for_state` returns only when the predicate is true, so we can remove the `FIXME`, and the IDLE check in following lines.